mypy/nodes.py

```diff
@@ -3146,10 +3146,10 @@ class FakeInfo(TypeInfo):
     def __init__(self, msg: str) -> None:
         self.msg = msg
 
-    def __getattribute__(self, attr: str) -> None:
+    def __getattribute__(self, attr: str) -> type:
         # Handle __class__ so that isinstance still works...
         if attr == "__class__":
-            return object.__getattribute__(self, attr)
+            return object.__getattribute__(self, attr)  # type: ignore[no-any-return]
```

Comment:
`__getattribute__` feels like it could be quite performance-sensitive, so `type: ignore` feels like the best option here imo.